examples/extsort/sort.go

```diff
@@ -0,0 +1,249 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/grandecola/bigqueue"
+)
+
+// TODO:
+//  - release resources for bigqueue
+//  - ensure that we do not cross the available memory
+//  - use fewer queues
+
+var (
+	bqFileIndexCount = 0
+)
+
+// ExternalSort perform external sort https://en.wikipedia.org/wiki/External_sorting
+// The inputPath should be a path to a file containing integers in each line
+// The outputPath is similarly formatted file with sorted integers
+// The tempPath is used to write intermediate files
+// maxMemSortSize is number of elements that can be sorted directly in memory
+func ExternalSort(inputPath, tempPath, outputPath string, maxMemSortSize int) error {
+	if err := os.RemoveAll(tempPath); err != nil {
```

Comment:
This is not required.

Reply:
this is required to ensure that we do not end up reading the previously created queues.

Reply:
I still think we should not delete all file in the directory. If someone specifies wrong tempPath, this can delete all files in that dir. We can create all queue dirs with specific prefix and delete all dirs starting with that prefix.

Reply:
fixed now